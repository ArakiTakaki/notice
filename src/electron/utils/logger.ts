import logger from "../../global/logger"

export const electronLogger = logger.child({ process: 'electron' })