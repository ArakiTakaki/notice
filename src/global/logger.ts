import pino from 'pino';
import { isBrowser, isPreload } from './constants';
const logger = pino({
  browser: {
    asObject: isBrowser || isPreload,
  }
});
const createProcess = (process: string) => ({ process })

export const createLoggerTag = (tag: string) => ({ tag })
export const electronLogger = logger.child(createProcess('electron'));
export const preloadLogger = logger.child(createProcess('preload'));
export const rendererLogger = logger.child(createProcess('renderer'));

export const createWebpackLogger = (tag: string) => logger.child(createProcess('webpack')).child(createLoggerTag(tag));
export const createWebpackDevServerLogger = (tag: string) => logger.child(createProcess('webpack-dev-server')).child(createLoggerTag(tag));

export default logger;
