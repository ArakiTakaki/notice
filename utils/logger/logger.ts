import { PATH } from "../../constants/dirPath";
import { resolve } from 'path';

export enum LOG_LEVEL {
  FATAL = 0,
  ERROR = 1,
  WARN = 2,
  INFO = 3,
  DEBUG = 4,
  TRACE = 5,
};

// INFO  2015-10-30 15:44:01,836 [http-8080-4] START - /path/to/example is Called from getApplication.
export const logger = (level: LOG_LEVEL, logName: string, logTarget: string, text: string, isOutput: boolean = false) => {
  const now = new Date();
  const date = `${now.getFullYear()}-${now.getMonth()}-${now.getDate()}`
  const time = `${now.getHours()}:${now.getMinutes()}:${now.getSeconds()},${now.getMilliseconds()}`;
  const status = (() => {
    switch (level) {
      case LOG_LEVEL.FATAL:
        return 'FATAL';
      case LOG_LEVEL.ERROR:
        return 'ERROR';
      case LOG_LEVEL.WARN:
        return 'WARN';
      case LOG_LEVEL.INFO:
        return 'INFO';
      case LOG_LEVEL.DEBUG:
        return 'DEBUG';
      case LOG_LEVEL.TRACE:
        return 'TRACE';
      default:
        return 'NONE';
    }
  })();
  const data = `${status} \t ${date} ${time} ${logName} ${logTarget} ${text}`;
  if (isOutput) {
    console.log(data);
    const fs = require('fs');
    fs.appendFile(resolve(PATH.LOG.OUTPUT_PATH, `${date}.log`), data);
  } else {
    console.log(data);
  }
  return data;
};
