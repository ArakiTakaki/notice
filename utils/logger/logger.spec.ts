import { logger } from "./logger";
import { LOG_LEVEL } from "./enum";
jest.mock('fs');

const cases = [
  LOG_LEVEL.FATAL,
  LOG_LEVEL.ERROR,
  LOG_LEVEL.WARN,
  LOG_LEVEL.INFO,
  LOG_LEVEL.DEBUG,
  LOG_LEVEL.TRACE,
]
describe('local test', () => {
  it('ALL_CASE', () => {
    cases.forEach(level => {
      const log = logger(level, 'SAMPLE_NAME', 'SAMPLE_TARGET', 'HOGE_TEXT');
      expect(log).toMatchSnapshot();
    });
  });

  it('OUTPUT_TEST', () => {
    cases.forEach(level => {
      const log = logger(level, 'SAMPLE_NAME', 'SAMPLE_TARGET', 'HOGE_TEXT', true);
      expect(log).toMatchSnapshot();
    });
  });
});