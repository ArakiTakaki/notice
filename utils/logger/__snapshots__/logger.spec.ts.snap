// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`local test ALL_CASE 1`] = `"FATAL 2020-4-31 0:3:44,123 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test ALL_CASE 2`] = `"ERROR 2020-4-31 0:3:44,134 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test ALL_CASE 3`] = `"WARN 2020-4-31 0:3:44,135 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test ALL_CASE 4`] = `"INFO 2020-4-31 0:3:44,136 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test ALL_CASE 5`] = `"DEBUG 2020-4-31 0:3:44,137 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test ALL_CASE 6`] = `"TRACE 2020-4-31 0:3:44,138 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test OUTPUT_TEST 1`] = `"FATAL 2020-4-31 0:3:44,139 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test OUTPUT_TEST 2`] = `"ERROR 2020-4-31 0:3:44,147 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test OUTPUT_TEST 3`] = `"WARN 2020-4-31 0:3:44,148 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test OUTPUT_TEST 4`] = `"INFO 2020-4-31 0:3:44,149 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test OUTPUT_TEST 5`] = `"DEBUG 2020-4-31 0:3:44,150 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;

exports[`local test OUTPUT_TEST 6`] = `"TRACE 2020-4-31 0:3:44,150 SAMPLE_NAME SAMPLE_TARGET HOGE_TEXT"`;
